File: imports-frontend-entities/src/entities/inspection_check.js
const handler = require('./base/handler')

module.exports = class InspectionCheck {

  constructor(obj = {}) {

    this.type = obj.type
    this.status = obj.status
    this.reason = obj.reason
    this.otherReason = obj.otherReason
    this.isSelectedForChecks = obj.isSelectedForChecks
    this.hasChecksComplete = obj.hasChecksComplete

    return Object.seal(new Proxy(this, handler))
  }
}

File: notification-schema-java/src/main/java/uk/gov/defra/tracesx/notificationschema/representation/InspectionCheck.java
package uk.gov.defra.tracesx.notificationschema.representation;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import uk.gov.defra.tracesx.notificationschema.representation.enumeration.CheckStatus;
import uk.gov.defra.tracesx.notificationschema.representation.enumeration.CheckType;

@Builder
@Data
@JsonInclude(Include.NON_EMPTY)
@NoArgsConstructor(access = AccessLevel.PUBLIC)
@AllArgsConstructor(access = AccessLevel.PUBLIC)
public class InspectionCheck {

  private CheckType type;
  private CheckStatus status;
  private String reason;
  private String otherReason;
  private Boolean isSelectedForChecks;
  private Boolean hasChecksComplete;
}
